src/test/java/org/mariadb/jdbc/failover/AuroraFailoverTest.java

```diff
@@ -119,21 +120,21 @@ public void pingReconnectAfterRestart() throws Throwable {
             int masterServerId = getServerId(connection);
             stopProxy(masterServerId);
 
-            long stoppedTime = System.currentTimeMillis();
             try {
                 st.execute("SELECT 1");
             } catch (SQLException e) {
                 //normal exception
             }
             restartProxy(masterServerId);
-            long restartTime = System.currentTimeMillis();
+            long restartTime = System.nanoTime();
 
             boolean loop = true;
             while (loop) {
                 if (!connection.isClosed()) {
                     loop = false;
                 }
-                if (System.currentTimeMillis() - restartTime > 15 * 1000) {
+                long duration = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - restartTime);
```

Comment:
I had considered that, let me know what your preference is.  You will see in the main code I did convert most to seconds.